src/riscv/lib/src/jit/state_access.rs

```diff
@@ -311,7 +311,7 @@ extern "C" fn f64_from_x64_unsigned_dynamic<MC: MemoryConfig>(
 extern "C" fn f64_from_x64_unsigned_static<RM: StaticRoundingMode, MC: MemoryConfig>(
     core: &mut MachineCoreState<MC, Owned>,
     xval: XValue,
-) -> f64 {
+) -> u64 {
```

Comment:
```suggestion
) -> FValue {
```

This needs to return an `FValue`.